packages/remix-server-runtime/errors.ts

```diff
@@ -57,14 +57,16 @@ export interface ThrownResponse<T = any> {
   data: T;
 }
 
-export interface SerializedError {
+// must be type alias due to inference issues on interfaces
+// https://github.com/microsoft/TypeScript/issues/15300
+export type SerializedError = {
   message: string;
   stack?: string;
-}
+};
 
-export async function serializeError(error: Error): Promise<SerializedError> {
+export async function serializeError(error: Error) {
   return {
     message: error.message,
     stack: error.stack,
-  };
+  } as SerializedError;
```

Comment:
Is this type cast necessary?
I think we could keep the `: Promise<SerializedError>` return value, no?